source/DG.Core/Scanners/ApplicationInstancesScanner.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace DG.Core.Scanners
+{
+    using System.Collections.Generic;
+
+    public class ApplicationInstancesScanner : IApplicationInstancesScanner
+    {
+        private readonly IConfigScanner configScanner;
+        private readonly string sectionName = "ApplicationInstances";
+        private readonly IDictionary<string, object> applicationInstancesConfiguration = new Dictionary<string, object>();
+
+        public ApplicationInstancesScanner(IConfigScanner configScanner)
+        {
+            this.configScanner = configScanner;
+        }
+
+        public void Initialize()
+        {
+            IDictionary<string, object> instancesData = this.configScanner.GetKeyValuesFromSection(this.sectionName);
+            var instances = instancesData.Values;
+            foreach (var instance in instances)
+            {
+                var instanceFields = (instance as IDictionary<string, object>).Values;
+                {
+                    foreach (var field_values in instanceFields)
+                    {
+                        IDictionary<string, object> instanceFieldsData = new Dictionary<string, object>();
+                        foreach (var field_value in (IDictionary<string, object>)field_values)
+                        {
+                            instanceFieldsData.Add(field_value.Key, field_value.Value);
+                        }
+
+                        var instanceType = instanceFieldsData["Type"];
```

Comment:
We can create variable
`private string instanceTypeFieldName = "Type"`
and use it here
`var instanceType = instanceFieldsData[instanceTypeFieldName];`
and later:
`return this.GetFieldValuesFromApps(instanceTypeFieldName);`
Same approach can be applied to `"Name"`

Reply:
I have reworked this part